mlflow/langchain/_langchain_autolog.py

```diff
@@ -180,6 +185,439 @@ def _chain_with_retriever(model):
     return False
 
 
+def get_mlflow_langchain_tracer():
+    from langchain.callbacks.base import BaseCallbackHandler
+    from langchain_core.agents import AgentAction, AgentFinish
+    from langchain_core.documents import Document
+    from langchain_core.load import dumpd
+    from langchain_core.messages import BaseMessage
+    from langchain_core.outputs import (
+        ChatGeneration,
+        ChatGenerationChunk,
+        GenerationChunk,
+        LLMResult,
+    )
+    from tenacity import RetryCallState
+
+    from mlflow.entities import ExceptionEvent, SpanEvent, SpanType
+    from mlflow.entities.span_status import SpanStatus
+    from mlflow.entities.trace_status import TraceStatus
+    from mlflow.tracing.types.wrapper import MLflowSpanWrapper
+
+    class MlflowLangchainTracer(BaseCallbackHandler, metaclass=ExceptionSafeAbstractClass):
+        """
+        Callback for auto-logging artifacts.
+        We need to inherit ExceptionSafeAbstractClass to avoid invalid new
+        input arguments added to original function call.
+        Args:
+            tracking_uri: MLflow tracking server uri.
+            experiment_name: Name of the experiment.
+            run_id: Id of the run to log the artifacts.
+        """
+
+        def __init__(self):
+            super().__init__()
+            self._mlflow_client = MlflowClient()
+            self._run_span_mapping: Dict[str, MLflowSpanWrapper] = {}
+
+        def _get_span_by_run_id(self, run_id: UUID) -> Optional[MLflowSpanWrapper]:
+            if span := self._run_span_mapping.get(str(run_id)):
+                return span
+            raise MlflowException(f"Span for run_id {run_id!s} not found.")
```

Comment:
Current rule of thumb is to raise an exception only if it is a fault of users e.g. invalid arguments passed to API, incorrect ID is specified, etc, and any internal error should only surfaced as warning not to block the main application. Do you think we can apply the same here?

Reply:
Yes make sense, though autologging will swallow all exceptions. I think if we don't want to throw exceptions here, is there a better way to end the callback? Cuz that means some methods are not called in order and the traces will be messed up

Reply:
Is there a way to end the callback from itself?

Reply:
Nope

Reply:
Ok, I'm fine with keeping it but let's monitor closely for autologging feedback. Afaik this happens more than we think normally....